docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -334,29 +252,261 @@ const theme = createTheme({
+    const ref = React.useRef();
+  - return <MyComponent innerRef={ref} />;
+  + return <MyComponent ref={ref} />;
+  }
+  ```
+
+#### withWidth
 
-- Renamed `fade` to `alpha` to better describe its functionality.
-  The previous name was leading to confusion when the input color already had an alpha value. The helper **overrides** the alpha value of the color.
+- This HOC was removed. There's an alternative using the `useMediaQuery` hook on [this page](/components/use-media-query/#migrating-from-withwidth).
 
-  ```diff
-  - import { fade } from '@material-ui/core/styles';
-  + import { alpha } from '@material-ui/core/styles';
+> At this point, you should see your application running without crash (some UI differences are expected). Next step is to go through component breaking changes below. In case that your application is still crash with unknown error, [open an issue](https://github.com/mui-org/material-ui/issues/new/choose) in material-ui repository.
 
-  const classes = makeStyles(theme => ({
-  -  backgroundColor: fade(theme.palette.primary.main, theme.palette.action.selectedOpacity),
-  +  backgroundColor: alpha(theme.palette.primary.main, theme.palette.action.selectedOpacity),
-  }));
-  ```
+## Handling breaking changes
 
-  You can use the [`fade-rename-alpha` codemod](https://github.com/mui-org/material-ui/tree/HEAD/packages/material-ui-codemod#fade-rename-alpha) for automatic migration.
+### Supported browsers and node versions
 
-- The `createStyles` function from `@material-ui/core/styles` was moved to the one exported from `@material-ui/styles`. It is necessary for removing the dependency to `@material-ui/styles` in the core package.
+The targets of the default bundle have changed.
+The exact versions will be pinned on release from the browserslist query `"> 0.5%, last 2 versions, Firefox ESR, not dead, not IE 11, maintained node versions"`.
 
-```diff
--import { createStyles } from '@material-ui/core/styles';
-+import { createStyles } from '@material-ui/styles';
-```
+The default bundle supports the following minimum versions:
+
+<!-- #stable-snapshot -->
+
+- Node 12 (up from 8)
+- Chrome 84 (up from 49)
+- Edge 85 (up from 14)
+- Firefox 78 (up from 52)
+- Safari 13 (macOS) and 12.2 (iOS) (up from 10)
+- and more (see [.browserslistrc (`stable` entry)](https://github.com/mui-org/material-ui/blob/HEAD/.browserslistrc#L11))
+
+It no longer supports IE 11.
+If you need to support IE 11, check out our [legacy bundle](/guides/minimizing-bundle-size/#legacy-bundle).
+
+### non-ref-forwarding class components
+
+Support for non-ref-forwarding class components in the `component` prop or as immediate `children` has been dropped. If you were using `unstable_createStrictModeTheme` or didn't see any warnings related to `findDOMNode` in `React.StrictMode` then you don't need to do anything.
+Otherwise check out the ["Caveat with refs" section in our composition guide](/guides/composition/#caveat-with-refs) to find out how to migrate.
```

Comment:
```suggestion
Otherwise check out the [Caveat with refs](/guides/composition/#caveat-with-refs) section in the composition guide to find out how to migrate.
```